vit-servicing-station-lib/src/db/queries/search/generic.rs

```diff
@@ -0,0 +1,46 @@
+use diesel::{
+    dsl::Like,
+    expression::bound::Bound,
+    query_dsl::{
+        methods::{FilterDsl, OrderDsl},
+        LoadQuery,
+    },
+    r2d2::{ConnectionManager, PooledConnection},
+    sql_types::Text,
+    Expression, TextExpressionMethods,
+};
+
+use crate::{db::DbConnection, v0::errors::HandleError};
+
+type SqlStr = Text;
+
+diesel::no_arg_sql_function!(RANDOM, (), "Represents the sql RANDOM() function");
```

Comment:
Not sure how I feel about non-deterministic/pure queries, to be honest. But I don't see a way to set up a seed neither, without defining a custom function or a table with a random sequence, that is.

I'd imagine this is to reduce biases? is it really that necessary?

Reply:
If I'm honest, I'm struggling to think up a use case for randomizing the results, it just said "random" on the ticket.

I guess because we currently don't support limits, the client could just randomize and we could drop this? A previous version of this did randomization in Rust, which is much easier to seed, but it still doesn't feel great, especially around scaling.